### PR TITLE
[T3A3][F12-A2]Lim Huan Hock

### PR DESCRIPTION
**Pull request by huanhock**
Implement classes

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -21,12 +27,14 @@
...
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = trimmedAddress;
+        addressArray = trimmedAddress.split(", ");
+        this.block = new Block(addressArray[0]);
+        this.street = new Street(addressArray[1]);
+        this.unit = new Unit(addressArray[2]);
+        this.postalCode = new PostalCode(addressArray[3]);
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java]: Is there anyway to improve readability? Say refactoring out magic numbers?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -38,19 +46,21 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        String stringAddress = block.getBlock() + ", " + street.getStreet() + ", " + 
+                unit.getUnit() + ", " + postalCode.getPostalCode();
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java]: Recall from the tutorial about the improvement that can be made to make this more readable.

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package seedu.addressbook.data.person;
+
+public class Block {
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Block.java]: Recall from our tutorial to add in class header comments.

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    private String block;
+    
+    public Block(String blockNumber){
+        this.block = blockNumber;
+        }
+        
+        public String getBlock() {
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Block.java]: Indentation should be done according to coding standard provided in the handbook.

File: test/input.txt [hunk 5]
```diff
@@ -40,7 +40,7 @@
   add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag
 
   # should add correctly and list non private information
-  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street
+  add Adam Brown p/111111 e/adam@gmail.com a/123, Streety MaccStreetStreet, #12-34, 123456
```
Comment (brandonyeoxg) [inline test/input.txt]: Glad to see test is updated =).

**Comment by huanhock** [comment]
Ready for review

**Review by brandonyeoxg — COMMENTED** [review]
@huanhock Some comments added. Please close the PR after reading comments. Do pay attention to indentation levels.

**Pull request closed without merging**